blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -0,0 +1,192 @@
+    copy(userAgent = userAgent)
+  def withUserAgent(userAgent: `User-Agent`): BlazeClientBuilder[F] =
+    withUserAgentOption(Some(userAgent))
+  def withoutUserAgent: BlazeClientBuilder[F] =
+    withUserAgentOption(None)
+
+  def withIdleTimeout(maxTotalConnections: Int): BlazeClientBuilder[F] =
+    copy(maxTotalConnections = maxTotalConnections)
+
+  def withMaxWaitQueueLimit(maxWaitQueueLimit: Int): BlazeClientBuilder[F] =
+    copy(maxWaitQueueLimit = maxWaitQueueLimit)
+
+  def withMaxConnectionsPerRequestKey(
+      maxConnectionsPerRequestKey: RequestKey => Int): BlazeClientBuilder[F] =
+    copy(maxConnectionsPerRequestKey = maxConnectionsPerRequestKey)
+
+  def withSslContextOption(sslContext: Option[SSLContext]): BlazeClientBuilder[F] =
+    copy(sslContext = sslContext)
+  def withSslContext(sslContext: SSLContext): BlazeClientBuilder[F] =
+    withSslContextOption(Some(sslContext))
+  def withoutSslContext: BlazeClientBuilder[F] =
+    withSslContextOption(None)
+
+  def withCheckEndpointAuthentication(checkEndpointIdentification: Boolean): BlazeClientBuilder[F] =
+    copy(checkEndpointIdentification = checkEndpointIdentification)
+
+  def withMaxResponseLineSize(maxResponseLineSize: Int): BlazeClientBuilder[F] =
+    copy(maxResponseLineSize = maxResponseLineSize)
+
+  def withMaxChunkSize(maxChunkSize: Int): BlazeClientBuilder[F] =
+    copy(maxChunkSize = maxChunkSize)
+
+  def withParserMode(parserMode: ParserMode): BlazeClientBuilder[F] =
+    copy(parserMode = parserMode)
+
+  def withBufferSize(bufferSize: Int): BlazeClientBuilder[F] =
+    copy(bufferSize = bufferSize)
+
+  def withExecutionContext(executionContext: ExecutionContext): BlazeClientBuilder[F] =
+    copy(executionContext = executionContext)
+
+  def withAsynchronousChannelGroupOption(
+      asynchronousChannelGroup: Option[AsynchronousChannelGroup]): BlazeClientBuilder[F] =
+    copy(asynchronousChannelGroup = asynchronousChannelGroup)
+  def withAsynchronousChannelGroup(
+      asynchronousChannelGroup: AsynchronousChannelGroup): BlazeClientBuilder[F] =
+    withAsynchronousChannelGroupOption(Some(asynchronousChannelGroup))
+  def withoutAsynchronousChannelGroup: BlazeClientBuilder[F] =
+    withAsynchronousChannelGroupOption(None)
+
+  def resource(implicit F: ConcurrentEffect[F]): Resource[F, Client[F]] =
+    Resource.make(
+      connectionManager.map(
+        manager =>
+          BlazeClient.makeClient(
+            manager = manager,
+            responseHeaderTimeout = responseHeaderTimeout,
+            idleTimeout = idleTimeout,
+            requestTimeout = requestTimeout
+        )))(_.shutdown)
+
+  def stream(implicit F: ConcurrentEffect[F]): Stream[F, Client[F]] =
+    Stream.resource(resource)
+
+  private def connectionManager(
+      implicit F: ConcurrentEffect[F]): F[ConnectionManager[F, BlazeConnection[F]]] = {
+    val http1: ConnectionBuilder[F, BlazeConnection[F]] = new Http1Support(
```

Comment:
We've made this client very HTTP/1 centric with this.  What would HTTP/2 for blaze even look like?